test/client.js

```diff
@@ -114,6 +115,9 @@ function buildServer (fastFlag) {
 server = buildServer().listen(port)
 
 describe('MqttClient', function () {
+  before(function (done) {
+    setTimeout(done, 10) // essentially set a 100ms buffer between each test
```

Comment:
Why is this needed? The overarching problem is that:

1. there is a shared broker. We should likely remove it.
2. there are a load of setTimeout or intervals both in the tests and in the code. This make the test suite very brittle

Reply:
@mcollina I am seeing in the tests that oftentimes the results of one test bleed over into another test causing the test suite to fail when individual tests will succeed. This was an attempt to alleviate some of the issues of this bleed over.